src/librustc_trans/glue.rs

```diff
@@ -235,56 +232,28 @@ impl<'tcx> DropGlueKind<'tcx> {
 
 fn get_drop_glue_core<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>,
                                 g: DropGlueKind<'tcx>) -> ValueRef {
-    debug!("make drop glue for {:?}", g);
     let g = g.map_ty(|t| get_drop_glue_type(ccx.tcx(), t));
-    debug!("drop glue type {:?}", g);
     match ccx.drop_glues().borrow().get(&g) {
-        Some(&glue) => return glue,
-        _ => { }
+        Some(&(glue, _)) => glue,
+        None => { bug!("Could not find drop glue for {:?} -- {} -- {}",
```

Comment:
Yes, it should have been predefined at this point. I'll add a comment.
After all the changes in this PR there is no HIR-walk trans anymore at all.